csrc/fusion.cpp

```diff
@@ -794,23 +794,32 @@ void Fusion::aliasOutputToInput(
     if (input->getDataType().value() != output->getDataType().value()) {
       output = castOp(input->getDataType().value(), output);
     }
-  }
 
-  NVF_ERROR(
-      isAliasCompatible(input, output),
-      "The input and output values are not alias-compatible.");
-  // Let integration hide any output that wasn't a fusion output when
-  // `aliasOutputToInput` was called. For example, running mean and var for
-  // batch norm.
-  io_alias_[output] = AliasInfo{
-      .type = type,
-      .aliased_io = input,
-      .hide_output = !output->isFusionOutput()};
-
-  // TODO: output should be marked at the end of fusion definition #1488
-  if (!output->isFusionOutput()) {
-    addOutput(output);
+    NVF_ERROR(
+        isAliasCompatible(input, output),
+        "The input and output values are not alias-compatible.");
+    // Let integration hide any output that wasn't a fusion output when
+    // `aliasOutputToInput` was called. For example, running mean and var for
+    // batch norm.
+    io_alias_[output] = AliasInfo{
+        .type = type,
+        .aliased_io = input,
+        .hide_output = !output->isFusionOutput()};
+
+    // TODO: output should be marked at the end of fusion definition #1488
+    if (!output->isFusionOutput()) {
+      addOutput(output);
+    }
+    return;
   }
+
+  // AllocationType::Evaluate
```

Comment:
```suggestion
```

The NVF_ERROR check below is pretty clear about what this region tries to handle.